include/benchmark/benchmark.h

```diff
@@ -611,7 +616,7 @@ struct State::StateIterator {
 
   BENCHMARK_ALWAYS_INLINE
   explicit StateIterator(State* st)
-      : cached_(st->max_iterations), parent_(st) {}
+      : cached_(st->error_occurred_ ? 0 : st->max_iterations), parent_(st) {}
```

Comment:
should this be a separate PR to keep the functional changes separate from the refactor?

Reply:
Yep. I'll commit separately.